src/sb-util.ts

```diff
@@ -255,6 +257,153 @@ export class BlockCollection implements Queryable {
+            if (opcode === BlockOpcodes.CONTROL_IF_ELSE) {
+                return {
+                    if: {
+                        condition: opBody,
+                    },
+                };
+            }
+
+            if (opcode === BlockOpcodes.CONTROL_WAIT_UNTIL) {
+                return {
+                    condition: opBody,
+                };
+            }
+        }
+
+        if (ControlBlocksArgs.SUBSTACK === substackType || ControlBlocksArgs.SUBSTACK2 === substackType) {
+            const substack =
+                ControlBlocksArgs.SUBSTACK2 === substackType
+                    ? ControlBlocksArgs.SUBSTACK2
+                    : ControlBlocksArgs.SUBSTACK;
+            if (opcode === BlockOpcodes.CONTROL_IF_ELSE) {
+                const substackBlockRef = this.getSubstackRef(block, substack);
+                const firstSubstackBlock = this.getBlockByID(substackBlockRef);
+                const body = this.render(firstSubstackBlock, []);
+                const then = substack === ControlBlocksArgs.SUBSTACK2 ? 'else' : 'then';
+                return {
+                    [then]: body,
+                };
+            }
+        }
+
+        console.warn(`${substackType} not implemented yet!`);
+        return {};
+    }
+
+    private render(block: Block, steps: any[], inputArgName?: string, fieldArgName?: string): string[] {
+        // STOP CONDITION
+        if (!block) {
+            return steps;
+        }
+
+        const opBody = this.getOpBody(block);
+        if (Object.keys(opBody).length === 0) {
+            steps.push(block.prop('opcode'));
+        } else {
+            steps.push(opBody);
+        }
+
+        // Follow the tree down condition and substack branches
+        const inputs = block.prop('inputs');
+        if (ControlBlocksArgs.CONDITION in inputs) {
+            const lastOpcode = steps.pop();
+            steps.push({
+                [lastOpcode]: this.getSubstackBody(block, ControlBlocksArgs.CONDITION),
+            });
+        }
+
+        if (ControlBlocksArgs.SUBSTACK in inputs) {
+            let substack = steps.pop();
+            if (typeof substack === 'string') {
+                substack = { [substack]: {} };
+            }
+            const lastOpcode = Object.keys(substack)[0];
+            Object.assign(substack[lastOpcode], this.getSubstackBody(block, ControlBlocksArgs.SUBSTACK));
+            steps.push(substack);
+        }
+
+        if (ControlBlocksArgs.SUBSTACK2 in inputs) {
+            let substack2 = steps.pop();
+            if (typeof substack2 === 'string') {
+                substack2 = { [substack2]: {} };
+            }
+            const lastOpcode = Object.keys(substack2)[0];
+            Object.assign(substack2[lastOpcode], this.getSubstackBody(block, ControlBlocksArgs.SUBSTACK2));
+            steps.push(substack2);
+        }
+
+        const nextBlock = this.getNextBlock(block);
```

Comment:
Why do we include the next block? Shouldn't this just return the block we are pointing at and its substacks and reporters?

Reply:
If we want it to be the following blocks in the collection we should make it the blocks in the collection and not the next block. The collection may be a filtered set of blocks and so not directly reference each other.

Reply:
As is this should print the remaining contents of a single script in a scratch sprite. If the sprite has many scripts, it won't include those even if they are in the collection.